processing/src/main/java/org/apache/druid/query/aggregation/JavaScriptAggregatorFactory.java

```diff
@@ -89,7 +95,7 @@ public JavaScriptAggregatorFactory(
   @Override
   public Aggregator factorize(final ColumnSelectorFactory columnFactory)
   {
-    checkAndCompileScript();
+    getCompiledScript();
```

Comment:
This method should be used instead of `compiledScript` a few lines below, that was the whole point. Same in other places where getCompiledScript() is called.